core/src/main/java/com/nisovin/magicspells/commands/VariableCommands.java

```diff
@@ -0,0 +1,158 @@
+import org.incendo.cloud.description.Description;
+import org.incendo.cloud.paper.PaperCommandManager;
+import org.incendo.cloud.component.CommandComponent;
+import org.incendo.cloud.parser.standard.StringParser;
+import org.incendo.cloud.suggestion.SuggestionProvider;
+import org.incendo.cloud.bukkit.data.SinglePlayerSelector;
+import org.incendo.cloud.bukkit.parser.selector.SinglePlayerSelectorParser;
+
+import org.bukkit.entity.Player;
+import org.bukkit.command.CommandSender;
+
+import io.papermc.paper.command.brigadier.CommandSourceStack;
+
+import com.nisovin.magicspells.Perm;
+import com.nisovin.magicspells.util.Util;
+import com.nisovin.magicspells.MagicSpells;
+import com.nisovin.magicspells.util.SpellData;
+import com.nisovin.magicspells.util.VariableMod;
+import com.nisovin.magicspells.variables.Variable;
+import com.nisovin.magicspells.util.managers.VariableManager;
+import com.nisovin.magicspells.variables.variabletypes.GlobalVariable;
+import com.nisovin.magicspells.variables.variabletypes.GlobalStringVariable;
+import com.nisovin.magicspells.commands.exceptions.InvalidCommandArgumentException;
+
+public class VariableCommands {
+
+	private static final CloudKey<SinglePlayerSelector> TARGET_PLAYER_KEY = CloudKey.of("player", SinglePlayerSelector.class);
+	private static final CloudKey<String> VARIABLE_MOD_KEY = CloudKey.of("variable mod", String.class);
+	private static final CloudKey<String> VARIABLE_KEY = CloudKey.of("variable", String.class);
+
+	static void register(@NotNull PaperCommandManager<CommandSourceStack> manager) {
+		Command.Builder<CommandSourceStack> base = manager.commandBuilder("ms", "magicspells").literal("variable");
+
+		var variableComponent = CommandComponent.<CommandSourceStack, String>builder()
+			.key(VARIABLE_KEY)
+			.parser(StringParser.stringParser())
+			.suggestionProvider(SuggestionProvider.suggestingStrings(
+				() -> MagicSpells.getVariableManager().getVariables().keySet().iterator()
+			));
+
+		manager.command(base
+			.literal("show")
+			.argument(variableComponent.description(Description.of("The variable to show.")))
+			.optional(TARGET_PLAYER_KEY, SinglePlayerSelectorParser.singlePlayerSelectorParser())
+			.permission(Perm.COMMAND_VARIABLE_SHOW)
+			.handler(VariableCommands::show)
+		);
+
+		Command.Builder<CommandSourceStack> modify = base
+			.literal("modify")
+			.argument(variableComponent.description(Description.of("The variable to modify.")))
+			.permission(Perm.COMMAND_VARIABLE_MODIFY);
+
+		manager.command(modify
+			.literal("*")
+			.required(
+				VARIABLE_MOD_KEY,
+				StringParser.greedyStringParser(),
+				Description.of("A variable modifier.")
+			)
+			.meta(HelpCommand.FILTER_FROM_HELP, true)
+			.handler(VariableCommands::modify)
+		);
+
+		manager.command(modify
+			.literal("-")
+			.required(VARIABLE_MOD_KEY, StringParser.greedyStringParser())
+			.meta(HelpCommand.FILTER_FROM_HELP, true)
+			.handler(VariableCommands::modify)
+		);
```

Comment:
I don't think we should use `*` to reference global variables. Historically, `-` was what we documented. There were also suggestions that we one day add a command to reset all variables of players, and I think `*` should be reserved for that.

The varNod parameter description needs to be copied/moved too.